### PR TITLE
Added RAdam optimizer

### PR DESCRIPTION
**Pull request by HareshKarnan**
From the authors original implementation : https://github.com/LiyuanLucasLiu/RAdam

Tested with MNIST and works without throwing errors.

### CODE REVIEW

**Comment by vincentqb** [comment]
We don't think this method should be in pytorch core, at least not yet.

Our reservation is that we want to include methods that the community uses as a standard, or else the code maintenance problem balloons up for us. We do show discretion based on what the paper shows as evidence, for example BatchNorm was included within weeks of it's publish date (in Torch). In terms of rejected methods, we've rejected newly minted papers such as Swish (#3260, #3182), [Yellowfin](https://github.com/pytorch/pytorch/issues/1960) and many others that haven't become standardized in the community (like LSTM/Transformer/BatchNorm).

If you have a different opinion, let us know why, and we can re-evaluate.

**Comment by Tony-Y** [comment]
[On the adequacy of untuned warmup for adaptive optimization](https://arxiv.org/abs/1910.04209)

This paper shows:

> the Rectified Adam (RAdam) algorithm can be characterized as four steps of momentum SGD, followed by Adam with a fixed warmup schedule. 

I made a Python package for warmup schedules described in the paper:

https://github.com/Tony-Y/pytorch_warmup
